packages/typespec-python/src/utils.ts

```diff
@@ -21,6 +22,9 @@ export function camelToSnakeCase(name: string): string {
             .replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`)
             .replace(/_+/g, "_");
 
+    if (!name.toLocaleLowerCase().includes("constructor")) {
+        return Style.snake(name, false);
```

Comment:
With reference of m4 which also use `Style` of package "@azure-tools/codegen".
I also find a very interesting point in TypeScript:
![image](https://github.com/Azure/autorest.python/assets/70930885/9073bb59-25d3-450a-8644-66902ca1ecdc)

"constrcutor"